### PR TITLE
update promtail daemonset docs

### PR DESCRIPTION
**Pull request by robbymilo**
**What this PR does / why we need it**:

I found it buried in the community site that you need to have the `HOSTNAME` env var set when using `kubernetes_sd_configs` in order for promtail to work.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: docs/sources/clients/promtail/installation.md [hunk 1]
```diff
@@ -78,6 +78,11 @@ spec:
         image: grafana/promtail
         args:
         - -config.file=/etc/promtail/promtail.yaml
+        env: 
+        - name: 'HOSTNAME', # needed when using kubernetes_sd_configs
```
Comment (kavirajk) [inline docs/sources/clients/promtail/installation.md:82]: Thanks @robbymilo :).

I can confirm we use `HOSTNAME` on [our cloud promtail config](https://github.com/grafana/loki/blob/main/tools/promtail.sh#L281-L284).

**Review by kavirajk — APPROVED** [review]
LGTM 👍 Thanks @robbymilo

**Review by KMiller-Grafana — APPROVED** [review]
LGTM

**Review by KMiller-Grafana — APPROVED** [review]
LGTM

**Review by owen-d — APPROVED** [review]

**Pull request merged**